evcouplings/align/tools.py

```diff
@@ -10,6 +10,304 @@
+            "--domT", str(domain_threshold),
+            "--incT", str(seq_threshold),
+            "--incdomT", str(domain_threshold)
+        ]
+    else:
+        cmd += [
+            "-E", str(seq_threshold),
+            "--domE", str(domain_threshold),
+            "--incE", str(seq_threshold),
+            "--incdomE", str(domain_threshold)
+        ]
+
+    # number of CPUs
+    if cpu is not None:
+        cmd += ["--cpu", str(cpu)]
+
+    # bias correction filter
+    if nobias:
+        cmd += ["--nobias"]
+
+    cmd += [hmmfile, database]
+
+    return_code, stdout, stderr = run(cmd)
+
+    # also check we actually created some sort of alignment
+    verify_resources(
+        "hmmsearch returned empty alignment: "
+        "stdout={} stderr={} file={}".format(
+            stdout, stderr, result.alignment
+        ),
+        result.alignment
+    )
+
+    return result
+
+
+
+def run_hmmbuild_and_search(**kwargs):
+                  
+    """
+    Search profile(s) against a sequence database.
+    Refer to HMMER documentation for details.
+
+    http://eddylab.org/software/hmmer3/3.1b2/Userguide.pdf
+
+    Parameters
+    ----------
+    Mandatory kwargs arguments are provided as arguments in
+    check_required function below.
+
+    Returns
+    -------
+    outcfg : dict
+        Output configuration of the protocol, including
+        the following fields:
+
+        * target_sequence_file
+        * sequence_file
+        * raw_alignment_file
+        * hittable_file
+        * focus_mode
+        * focus_sequence
+        * segments
+
+    Raises
+    ------
+    ExternalToolError, ResourceError
+    """
+
+    check_required(
+        kwargs,
+        [
+            "prefix", "msafile", 
+            "cpu", "database", "nobias",
+            "use_bitscores", "domain_threshold",
+            "seq_threshold", "hmmbuild",
+            "hmmsearch"
+        ]
+    ) 
+    
+    prefix = kwargs["prefix"]
+    msafile = kwargs["msafile"] 
+    cpu = kwargs["cpu"]
+    database = kwargs["database"]
+    nobias = kwargs["nobias"]
+    use_bitscores = kwargs["use_bitscores"]
+    domain_threshold = kwargs["domain_threshold"] 
+    seq_threshold = kwargs["seq_threshold"]
+    hmmbuild = kwargs["hmmbuild"]
+    hmmsearch = kwargs["hmmsearch"]
+
+
+    verify_resources(
+        "Input file does not exist or is empty",
+        database
+    )
+
+    # path to the directory to save the hmmfile in
+    hmmfile = prefix + ".hmm"
+
+    # create a hmmfile from the given msa file
+    # if hmmfile doesn't exist
+    if not path.isfile(hmmfile):
+        run_hmmbuild(
+            msafile, prefix, cpu,
+            binary=hmmbuild
+        )
+
+
+    # running hmmsearch binary with the hmmfile
+    result = run_hmmsearch(
+                hmmfile, database, prefix,
+                use_bitscores, domain_threshold,
+                seq_threshold, nobias, cpu,
+                binary=hmmsearch
+             )
+
+    return result
```

Comment:
The result that is returned needs to include the hmmfile that was created, ie it should be a union of HmmBuildResult and HmmsearchResult